playbooks/roles/openstack-remove-compute/tasks/main.yml

```diff
@@ -56,24 +55,22 @@
   until: nova_pod_name.stdout | length | int == 0
   retries: 30
   delay: 10
-  delegate_to: "{{ groups['soc-deployer'][0] }}"
+  when: inventory_hostname == groups['soc-deployer'][0]
   failed_when: false
   changed_when: false
 
 - name: Delete openstack compute service.
   command: "openstack compute service delete {{ compute_id.stdout }}"
   environment:
     OS_CLOUD: openstack
-  delegate_to: "{{ groups['soc-deployer'][0] }}"
-  when: compute_id.stdout | length | int != 0
+  when: inventory_hostname == groups['soc-deployer'][0] and compute_id is defined and compute_id.stdout | length | int != 0
   failed_when: false
 
 - name: Delete subvolumes from compute node.
   import_role:
     name: airship-configure-worker
     tasks_from: subvolumes-setup.yml
-  delegate_to: '{{ compute_node_name }}'
+  when: inventory_hostname == compute_node_name
```

Comment:
just occurred to me that since we are counting on the compute host name in openstack and in ansible inventory to be identical, we should call it out in the deployment guide inventory file section. can be different doc commit though,

Reply:
Ok will update the doc.

Reply:
May be add this in same PR so that it can be reviewed in same place.

Reply:
done